### DISCUSSION TITLE
Accessing repeated elements

### DISCUSSION THREAD

**Discussion by gerum100** [slint-ui/slint#6689] [Q&A]
I try to write a slint gui that does some custom layouting, until know I have this code (the position computations are more complicated, but the relevant thing is that they depend on the previous elements):

```slint
import { Button } from "std-widgets.slint";

export component AppWindow inherits Window {
    b1 := Button {
        text: "B1";
        x: 0;
        y: 0;
    }
    b2 := Button {
        text: "B2";
        x: b1.x + b1.width;
        y: b1.y + b1.height;
    }
    b3 := Button {
        text: "B3";
        x: b2.x + b2.width;
        y: b2.y + b2.height;
    }
}
```

An this does in principle work as intended, but this allows only to have a compile-time fixed number of elements in the layout, but what I need is a dynamic number of elements.

So something like this (which does not work) is desired:
```slint
export component AppWindow inherits Window {
    for x[index] in 5: b := Button {
        text: "B1";
        x: b[index-1].x + b[index-1].width;
        y: b[index-1].y + b[index-1].height;
    }
}
```
Does anybody know how to access repeated elements in such a way?

I do not ask this as a feature request, because you can already give repeated elements a name, so I assume that there is a possibility to do things like this.

**Comment by aplantier** [comment]
For my understanding it is not possible. I have the feeling it might be in the future, if it won't i really don't understand the meaning of the Loop-Reapeated component  id.

Edit: I think it was to obvious, you might want to have spaces not equals  ( :`   solution not related then     

~~But for your question, it seems the space between you element is always the same.~~
~~The following solution would work for you ?~~

~~slint
in-out property<length> width_space:20px;
in-out property<length> height_space:20px;
    for but[index] in 5: b := Button {
        text: "B1";
        x: 0 /*starting x point*/ +index *width_space;
        y: 0/*starting y point*/ +index*height_space;
    }
```~~
